x-pack/plugins/observability/public/components/shared/alert_search_bar/alert_search_bar.tsx

```diff
@@ -27,84 +26,79 @@ const getAlertStatusQuery = (status: string): Query[] => {
 
 export function ObservabilityAlertSearchBar({
   appName,
+  defaultSearchQueries = DEFAULT_QUERIES,
+  onEsQueryChange,
+  onKueryChange,
+  onRangeFromChange,
+  onRangeToChange,
+  onStatusChange,
+  kuery,
   rangeFrom,
-  setRangeFrom,
   rangeTo,
-  setRangeTo,
-  kuery,
-  setKuery,
   status,
-  setStatus,
-  setEsQuery,
-  queries = DEFAULT_QUERIES,
-}: AlertSearchBarProps) {
-  const {
-    data: {
-      query: {
-        timefilter: { timefilter: timeFilterService },
-      },
-    },
-    notifications: { toasts },
-    triggersActionsUi: { getAlertsSearchBar: AlertsSearchBar },
-  } = useKibana<ObservabilityAppServices>().services;
+}: ObservabilityAlertSearchBarProps) {
+  const { AlertsSearchBar, errorToast, timeFilterService } = useServices();
 
-  const onStatusChange = useCallback(
+  const onAlertStatusChange = useCallback(
     (alertStatus: AlertStatus) => {
-      setEsQuery(
+      onEsQueryChange(
         buildEsQuery(
           {
             to: rangeTo,
             from: rangeFrom,
           },
           kuery,
-          [...getAlertStatusQuery(alertStatus), ...queries]
+          [...getAlertStatusQuery(alertStatus), ...defaultSearchQueries]
         )
       );
     },
-    [kuery, queries, rangeFrom, rangeTo, setEsQuery]
+    [kuery, defaultSearchQueries, rangeFrom, rangeTo, onEsQueryChange]
   );
 
   useEffect(() => {
-    onStatusChange(status);
-  }, [onStatusChange, status]);
+    onAlertStatusChange(status);
+  }, [onAlertStatusChange, status]);
 
-  const onSearchBarParamsChange = useCallback(
+  const onSearchBarParamsChange = useCallback<
+    (query: {
+      dateRange: { from: string; to: string; mode?: 'absolute' | 'relative' };
+      query: string;
+    }) => void
+  >(
     ({ dateRange, query }) => {
       try {
         // First try to create es query to make sure query is valid, then save it in state
         const esQuery = buildEsQuery(
           {
-            to: rangeTo,
-            from: rangeFrom,
+            to: dateRange.to,
+            from: dateRange.from,
           },
           query,
-          [...getAlertStatusQuery(status), ...queries]
+          [...getAlertStatusQuery(status), ...defaultSearchQueries]
         );
-        setKuery(query);
+        onKueryChange(query);
         timeFilterService.setTime(dateRange);
-        setRangeFrom(dateRange.from);
-        setRangeTo(dateRange.to);
-        setEsQuery(esQuery);
+        onRangeFromChange(dateRange.from);
+        onRangeToChange(dateRange.to);
+        onEsQueryChange(esQuery);
       } catch (error) {
-        toasts.addError(error, {
+        errorToast(error, {
           title: i18n.translate('xpack.observability.alerts.searchBar.invalidQueryTitle', {
             defaultMessage: 'Invalid query string',
           }),
         });
-        setKuery(DEFAULT_QUERY_STRING);
+        onKueryChange(DEFAULT_QUERY_STRING);
```

Comment:
You have `onKueryChange` set twice.  Surprised you didn't get an `eslint` warning about overriding props inline.